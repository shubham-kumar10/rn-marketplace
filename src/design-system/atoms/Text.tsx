import React from 'react';
import { Text as RNText, TextProps, StyleSheet } from 'react-native';
import theme from '../theme';

interface Props extends TextProps {
  variant?: keyof typeof theme.typography.textStyles;
  color?: keyof typeof theme.colors.text;
}

const Text: React.FC<Props> = ({
  variant = 'body',
  color = 'primary',
  style,
  ...props
}) => {
  return (
    <RNText
      style={[
        styles.text,
        theme.typography.textStyles[variant],
        { color: theme.colors.text[color] },
        style,
      ]}
      {...props}
    />
  );
};

const styles = StyleSheet.create({
  text: {
    fontFamily: theme.typography.fontFamily.regular,
  },
});

export default Text;
