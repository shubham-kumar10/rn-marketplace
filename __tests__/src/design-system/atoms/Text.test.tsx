import React from 'react';
import { render } from '@testing-library/react-native';
import Text from '../../../../src/design-system/atoms/Text';

describe('Text', () => {
  it('renders correctly with default props', () => {
    const { toJSON } = render(<Text>Default Text</Text>);
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders correctly with body variant', () => {
    const { toJSON } = render(<Text variant="body">Body Text</Text>);
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders correctly with heading variant', () => {
    const { toJSON } = render(<Text variant="heading">Heading Text</Text>);
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders correctly with custom color', () => {
    const { toJSON } = render(<Text color="primary">Colored Text</Text>);
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders correctly with custom style', () => {
    const { toJSON } = render(
      <Text style={{ fontSize: 20 }}>Styled Text</Text>,
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
