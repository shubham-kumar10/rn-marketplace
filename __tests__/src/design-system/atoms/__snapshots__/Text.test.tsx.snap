// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text renders correctly with body variant 1`] = `
<Text
  style={
    [
      {
        "fontFamily": "System",
      },
      {
        "fontSize": 16,
        "fontWeight": "400",
        "lineHeight": 20,
      },
      {
        "color": "#212121",
      },
      undefined,
    ]
  }
>
  Body Text
</Text>
`;

exports[`Text renders correctly with custom color 1`] = `
<Text
  style={
    [
      {
        "fontFamily": "System",
      },
      {
        "fontSize": 16,
        "fontWeight": "400",
        "lineHeight": 20,
      },
      {
        "color": "#212121",
      },
      undefined,
    ]
  }
>
  Colored Text
</Text>
`;

exports[`Text renders correctly with custom style 1`] = `
<Text
  style={
    [
      {
        "fontFamily": "System",
      },
      {
        "fontSize": 16,
        "fontWeight": "400",
        "lineHeight": 20,
      },
      {
        "color": "#212121",
      },
      {
        "fontSize": 20,
      },
    ]
  }
>
  Styled Text
</Text>
`;

exports[`Text renders correctly with default props 1`] = `
<Text
  style={
    [
      {
        "fontFamily": "System",
      },
      {
        "fontSize": 16,
        "fontWeight": "400",
        "lineHeight": 20,
      },
      {
        "color": "#212121",
      },
      undefined,
    ]
  }
>
  Default Text
</Text>
`;

exports[`Text renders correctly with heading variant 1`] = `
<Text
  style={
    [
      {
        "fontFamily": "System",
      },
      {
        "fontSize": 24,
        "fontWeight": "700",
        "lineHeight": 24,
      },
      {
        "color": "#212121",
      },
      undefined,
    ]
  }
>
  Heading Text
</Text>
`;
